src/UniswapWormholeMessageReceiver.sol

```diff
@@ -73,13 +77,15 @@ contract UniswapWormholeMessageReceiver {
      * @param _messageSender Address of the UniswapWormholeMessageSender contract on ethereum in Wormhole format,
      * i.e. 12 zero bytes followed by a 20-byte Ethereum address.
      */
-    constructor(address wormholeAddress, bytes32 _messageSender) {
+    constructor(address wormholeAddress, bytes32 _messageSender, uint16 _chainId) {
         // sanity check constructor args
         require(wormholeAddress != address(0), "Invalid wormhole address");
         require(_messageSender != bytes32(0) && bytes12(_messageSender) == 0, "Invalid sender contract");
+        require(_chainId != uint16(2), "Invalid chain id, receiver should not be deployed on Ethereum");
```

Comment:
is there a reason not to use `ETHEREUM_CHAIN_ID` in the comparison here (for clarity) instead of `uint16(2)`

Reply:
yep, agreed, fixed.

Reply:
nit: should we have an additional sanity check that ensures `_chainId` isn't 0?